### ISSUE TITLE
The `assert` command line flag should be properly documented

### ISSUE THREAD

**Issue by blacksun1** [hapijs/lab#634]
**tldr:** Unless you use the Lab `assert` command line flag it will ignore the fact that your Code tests are not complete

---

Hi,

I've only just found the `--asert` flag after using Lab and Code for months and I feel that it should be pointed out more in the documentation. 

For instance, If I ran Lab over the following assertions

```js
const Lab = require('lab');
const Code = require('code');

const lab = exports.lab = Lab.script();

lab.describe("expectation", () => {

  lab.it("should call the true method", done => {

    Code.expect(true).to.be.true;

    return done();
  });

  lab.it("does call the true method", done => {

    Code.expect(true).to.be.true();

    return done();
  });

});
```

If I use the `--assert code` flag, the code will fail with the error

```
Incomplete assertion at /home/vagrant/repos/eg-dashboard-services/test-bin/brokenTest.js:22.22

There were 1 test script error(s).
```

which is fantastic and what I should have expected after reading the acknowledgements of the [code README](https://github.com/hapijs/code/blob/master/README.md#acknowledgments) such as

> This will alert when a statement like expect(5).to.be.a.string is not allowed to remain unnoticed (and fail to throw due to the missing ()).

I think it's great that it works with multiple assertion libraries but somewhere in either the Code or Lab documentation it should be loud and clear that this is how it should work.

BTW, here is the original issue that probably caused this behaviour in the first place: https://github.com/hapijs/lab/issues/255

**Comment by geek** [comment]
@blacksun1 thanks for raising this. I am happy to update our documentation to make this clearer. Is this something that you want to take a run at updating?

**Comment by blacksun1** [comment]
Hey @geek,

I would be quite happy if I knew what the rules were around it. What does this flag do besides allow unfinished assertions when used with the flag of Code? Does it work with any other assertion libraries? Is there a certain interface to make it work?

Regards,

**Comment by blacksun1** [comment]
I've put something together just to make sure that my understanding is correct. So, Lab.assertions will then become code. https://gist.github.com/blacksun1/9c1f5c6d1e2a72227d71e96f13f577b8

Anything else I'm missing?

**Comment by geek** [comment]
@blacksun1 that is really all it does.

**Comment by blacksun1** [comment]
OK, I'll do something in the next week or so.

**Comment by blacksun1** [comment]
@geek Please see pull request #639
